### PR TITLE
Queue | Don't wipe issue dispositions during checkout flow

### LINKED ISSUE #5396: Persist issue dispositions in the attorney check out flow
An attorney received a case that his judge asked him to make revisions on. So, he had already entered issue dispositions in his initial check out to the judge. Upon starting the check out flow again after he got the case back, all of those dispositions were cleared. These issue dispositions should persist so the attorney does not have to do double work. This is like how we persist the attorney's issue dispositions for judge check out.

## AC
- If a case already has issue dispositions selected and the attorney initiates check out flow, those issue dispositions should be pre-populated.

### PR DESCRIPTION
**Pull request by dannysellers**
Resolves #5396 

### Description
When an attorney goes to the Draft Decision checkout flow, we un-set issue dispositions, forcing them to set options in the SelectDispositionsView. Sometimes, though, a judge will check out a case to an attorney for review, in which case its issues already have dispositions. If applicable, they should render in the SelectDispositionsView.

### Acceptance Criteria 
- [ ] Go to Draft Decision checkout flow for an appeal with issues with dispositions
- [ ] Confirm disposition dropdowns have those values pre-selected on page load

### Testing Plan
1. Go to `/queue` as `BVAAABSHIRE` (i.e. to view cases that have already been checked out / had dispositions set), start a Draft Decision checkout flow
2. Confirm disposition dropdowns have pre-set values
3. Finalize Draft Decision checkout, updating dispositions, check out to Anjali Abshire (`BVAAABSHIRE`). After returning to `/queue`, reload*, navigate to that appeal, confirm dispositions were updated

disposition_dropdowns

*Reloading is needed here because when finishing a checkout, we take a shortcut and simply delete the appeal from redux, anticipating/mimicking the server response (i.e. checking out to another user). In this case, we're checking an appeal out to the same user, though, so the appeal disappears until the page is reloaded

### CODE REVIEW

**Comment by dannysellers** [comment]
One open question is what to do with non-standard / legacy dispositions?

Of [all possible dispositions](https://github.com/department-of-veterans-affairs/caseflow/blob/master/constants/VACOLS_DISPOSITIONS_BY_ID.json), we currently only display 1 through 9, but should support displaying other dispositions for backwards compatibility.

This isn't a problem for `IssueListItem` (the static component that displays issue dispositions) but there is a complication within the Select Dispositions View: within each `SelectIssueDispositionDropdown`, the option displayed must be a choosable option, or else it won't show up correctly, and we'll end up forcing the user (through validation)  to choose one of the 1–9 dispositions.

[Previous discussion](https://github.com/department-of-veterans-affairs/caseflow/issues/5396#issuecomment-386691575)

File: client/app/queue/constants.js [hunk 1]
```diff
@@ -75,7 +76,7 @@ export const SEARCH_ERROR_FOR = {
 
 export const CASE_DISPOSITION_ID_BY_DESCRIPTION = Object.assign({},
   ...Object.keys(VACOLS_DISPOSITIONS_BY_ID).map((dispositionId) => ({
-    [VACOLS_DISPOSITIONS_BY_ID[dispositionId].toLowerCase().replace(/ /g, '_')]: dispositionId
```
Comment (dannysellers) [inline client/app/queue/constants.js]: This wouldn't remove punctuation, as in `Dismissed, Death`, resulting in `dismissed,_death` as the key (compared to `dismissed_death` coming from the server)

File: client/app/util/StringUtil.js [hunk 2]
```diff
@@ -64,6 +64,13 @@ const StringUtil = {
   },
   html5CompliantId(str) {
     return str.replace(/[^A-Za-z0-9-]/g, '-').replace(/-+/g, '-');
+  },
+
+  // convert (Capitalized) Words to lowercase, snake_case, remove punctuation
```
Comment (dannysellers) [inline client/app/util/StringUtil.js]: This mimics how we transform disposition labels when [loading issues from a VACOLS hash](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/issue.rb#L222)

File: client/app/queue/components/IssueRemandReasonsOptions.jsx [hunk 3]
```diff
@@ -256,7 +256,7 @@ const mapStateToProps = (state, ownProps) => {
 
   return {
     appeal,
-    issues: _.filter(issues, (issue) => issue.disposition === 'Remanded'),
+    issues: _.filter(issues, (issue) => issue.disposition === 'remanded'),
```
Comment (lowellrex) [inline client/app/queue/components/IssueRemandReasonsOptions.jsx]: Does it makes sense to add "remanded" to `queue/constants.js`?

File: client/app/util/StringUtil.js (hunk 2 above)
Comment (lowellrex) [reply client/app/util/StringUtil.js]: I think `parameterize()` might not handle the case of "Advance Withdrawn by Appellant/Rep" correctly:
```
js> "Advance Withdrawn by Appellant/Rep".toLowerCase().replace(/ /g, '_').replace(/,/g, '');
"advance_withdrawn_by_appellant/rep"

rails c> "Advance Withdrawn by Appellant/Rep".parameterize.underscore
"advance_withdrawn_by_appellant_rep"
```

Perhaps something like the following might be a little more precise?
```
parameterize(str) {
  return str.toLowerCase().replace(/\W/g, '_');
}
```

**Comment by lowellrex** [comment]
Also, do we want to make this change for the attorneys' view as well? Will that be a part of a different ticket?

**Pull request merged**